openverse_catalog/dags/providers/provider_api_scripts/metropolitan_museum.py

```diff
@@ -6,138 +6,167 @@
+                logger.info(f"Retrieved {self.non_cc0_objects} non-CC0 records.")
+            return None
+
+        main_image = object_json.get("primaryImage")
+        other_images = object_json.get("additionalImages", [])
+        image_list = [main_image] + other_images
+
+        meta_data = self._get_meta_data(object_json)
+        raw_tags = self._get_tag_list(object_json)
+        title = self._get_title(object_json)
+        artist = self._get_artist_name(object_json)
+
+        # We aren't currently populating creator_url. In theory we could url encode
+        # f"https://collectionapi.metmuseum.org/public/collection/v1/search?artistOrCulture={artist}"
+        # per API guide here: https://metmuseum.github.io/#search
+        # but it seems fairly buggy (i.e. nonresponsive), at least when tested with
+        # "Chelsea Porcelain Manufactory" and "Minton(s)" and "Jean Pucelle"
+        # Should we use artistWikidata_URL or artistULAN_URL? They're populated approx
+        # 65% of the time.
+
+        return [
+            {
+                "foreign_landing_url": object_json.get("objectURL"),
+                "image_url": img,
+                "license_info": self.DEFAULT_LICENSE_INFO,
+                "foreign_identifier": self._get_foreign_id(object_id, img),
+                "creator": artist,
+                "title": title,
+                "meta_data": meta_data,
+                "raw_tags": raw_tags,
+            }
+            for img in image_list
+        ]
+
+    def _get_foreign_id(self, object_id: int, image_url: str):
+        unique_identifier = image_url.split("/")[-1].split(".")[0]
+        return f"{object_id}-{unique_identifier}"
+
+    def _get_meta_data(self, object_json):
+        if object_json is None:
+            return
+        if object_json.get("accessionNumber"):
+            return {
+                "accession_number": object_json.get("accessionNumber"),
+            }
+
+    def _get_tag_list(self, object_json):
+        if object_json is None:
+            return
+        tag_list = [
+            tag
+            for tag in [
+                object_json.get("department"),
+                object_json.get("medium"),
+                object_json.get("culture"),
+                object_json.get("objectName"),
+                self._get_artist_name(object_json),
+                object_json.get("classification"),
+                object_json.get("objectDate"),
+                object_json.get("creditLine"),
+                object_json.get("period"),
+            ]
+            if tag
+        ]
+        if object_json.get("tags"):
+            tag_list += [tag["term"] for tag in object_json.get("tags")]
+        return tag_list
+
+    def _get_title(self, record):
+        if record is not None:
+            # Use or to skip false-y (empty) titles: ""
+            return record.get("title") or record.get("objectName")
+
+    def _get_artist_name(self, record):
```

Comment:
Yes, I just tried to clarify. In the process it occurred to me that we could consider each object a separate "batch", but my understanding is that there are a bunch that only have one record and I don't know what I would call the initial list of objects then, so I'm leaving it as is otherwise.

Reply:
Oh wow -- 'object' vs 'record' in this context _is_ confusing, I can see what you mean. Thanks for adding the comment, that's very clear now!